File: leetcode/1768_MergeStringsAlternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        
        int longWordLen = 0;
        if (word1.length() > word2.length()) longWordLen = word1.length();
        else longWordLen = word2.length();

        // abc - xyz -> axbycz

        String outputString = "";
    
        for (int i = 0; i < longWordLen; i++){

            if (word1.length() > i) outputString += word1.charAt(i);
            if (word2.length() > i) outputString += word2.charAt(i);
        }

        return outputString;
    }
}

File: leetcode/1768_MergeStringsAlternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:

        newLen = max(len(word1), len(word2))

        resultString = ""

        for i in range(newLen):
            
            if len(word1) > i:
                resultString += word1[i]
            if len(word2) > i:
                resultString += word2[i]
        
        return resultString

        
'''
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        merged = []
        maxLength = max(len(word1), len(word2))
        for i in range(maxLength):
            if i < len(word1):
                merged += word1[i]
            if i < len(word2):
                merged += word2[i]

        return "".join(merged)
'''
        
        